Chocolatey package grunt task

scowalt opened the pull request:
This resolves #54 

Let me know if you want anything changed!

bastimeyer commented:
There are some things that I think should be changed (in addition to a final commit squeeze).
But thanks already!!!

- [ ] [The project description has been changed recently.](https://github.com/bastimeyer/livestreamer-twitch-gui/pull/66/files#diff-ce1d77f3462c23f1f773fc15a2caf355R10)  
  See [here](https://github.com/bastimeyer/livestreamer-twitch-gui/commit/e43ff0c2def1450d23648cb1d889131906307213#diff-04c6e90faac2675aa89e2176d2eec7d8R15)...
- [ ] [Release notes parsing improvement (will comment on diff)](https://github.com/bastimeyer/livestreamer-twitch-gui/pull/66/files#diff-c647349bca61255be97c141d3fde92dfR24)
- [ ] Should there also be a `grunt clean` task target? I don't know.

bastimeyer commented on build/tasks/common/package.js:
```diff
@@ -0,0 +1,31 @@
+				return release !== null;
+			});
+
+		for ( var index in releases ) {
```
This can be done without an additional for loop (which also makes it much more readable).
```js
			.filter(function( release ) {
				return release !== null
				    && release[1] === version;
			});

		if ( releases.length !== 1 || !releases[0][5] ) {
			throw new Error( "Release notes for '" + version + "' not found." );
		}

		return releases[0][5].trim();
  ```

scowalt commented:
Hello again!
* I updated the application description in the template (including links as formatted in the README)
* Thanks for the code improvement on the release notes parsing! I included your changes in my pull request
* I added the task `grunt clean:package_chocolatey` that cleans out the package folder. It makes sense to have this task included, hopefully my naming follows the proper conventions.

bastimeyer commented:
Looks good, thanks!
The clean task should also be included in the task list here:
https://github.com/bastimeyer/livestreamer-twitch-gui/pull/66/files#diff-c640a975fa9aa821b9982084275e0d2cR15

But apart from that (and a commit squeeze), I think it's ready to get merged.

scowalt commented:
Wouldn't adding `clean:package_chocolatey` to that list simply remove all of the files immediately after the package is created? Or should the clean command be removing all of chocolatey files *except* for the the compiled `.nupkg` file?

Also, should I squeeze my commits using [this method](http://stackoverflow.com/a/22959708/1222411) or is there a better one you would recommend? (I haven't done a commit squeeze before)

bastimeyer commented:
No, I meant running the clean task first, then the other two, so the folder is empty before generating the files.

> Also, should I squeeze my commits using this method or is there a better one you would recommend? (I haven't done a commit squeeze before)

Yes, right. Also don't forget to force push to your remote branch afterwards.

scowalt commented:
> No, I meant running the clean task first

\*facepalm\* right, that makes much more sense. Not sure why I didn't think about it that way

scowalt commented:
Alright, I've added the clean task to the packaging process and squashed my commits.

bastimeyer commented:
Thank you!
